src/main/java/seedu/mark/model/bookmark/Folder.java

```diff
@@ -0,0 +1,55 @@
+package seedu.mark.model.bookmark;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.mark.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Folder in Mark.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidFolder(String)}
+ */
+public class Folder {
+    public static final String MESSAGE_CONSTRAINTS = "Folder names should be alphanumeric";
```

Comment:
Why alphanumeric? So underscore is not allowed?

Reply:
No reason, other than underscore should we allow other characters?

Reply:
I'll leave this for another time, i filed an issue for now.